### ISSUE TITLE
Order of concatenation

### ISSUE THREAD

**Issue by dionysiosarvanitis** [laravel-mix/laravel-mix#2985]
-   Laravel Mix Version: 6.0.19 (`npm list --depth=0`)
-   Node Version (`node -v`): 14.16.1
-   NPM Version (`npm -v`): 6.14.12
-   OS: Windows 10

### Description:

Code example below generates `styles.css` file that contains vue extracted css merged with application css coming from `master.scss`. Since my latest update, the order of concatenation has changed placing vue extracted styles after application's styles. Is this the intented behaviour? Is there any other way to specify the order in order to stop facing specificity issues.

```js
    mix.vue({
        extractStyles: true,
    })
    .sass( 'master.scss', 'styles.css')
    .extract()
```

**Comment by mikerogerz** [comment]
@dionysiosarvanitis Where you able to find a solution? Running into the same issue since upgrading to version 6 (from 5).

**Comment by thecrypticace** [comment]
This was an intended change. The behavior previously was more or less undefined. We should be able to control this though.

**Comment by dionysiosarvanitis** [comment]
I'm not sure what initially went wrong. I was suddenly able to build successfully. My current configuration is:

```js
mix.options({
    notifications: true,
    autoprefixer: {remove: false},
    processCssUrls: true,
    terser: {
        terserOptions: {
            warnings: false,
        },
    },
    legacyNodePolyfills: true,
});

mix.extend('vuetify', (webpackConfig) =>
    webpackConfig.plugins.push(
        new VuetifyLoaderPlugin(),
    ),
);

mix.copyDirectory('resources/images', 'public/images')
    .vuetify()
    .js('resources/js/app.js', global.Config.publicPath + '/js')
    .vue({
        version: 2,
        extractStyles: true,
    })
    .extract()
    .version()
    .sass('resources/sass/master.scss', global.Config.publicPath + '/css/styles.css')
    .sourceMaps(false, 'source-map');
```

**Comment by mikerogerz** [comment]
Based on my testing, the order is correct with:
`extractStyles: true`

But not using a defined path, such as in my case:
`extractStyles: 'static/landing/css/landing.css'`

In the second configuration, global styles are loaded after the component styles (instead of before in version 5).

**Comment by thecrypticace** [comment]
@mikerogerz Is `static/landing/css/landing.css` an existing file with other CSS placed in it?

**Comment by mikerogerz** [comment]
@thecrypticace It's the location to output the generated css to. The styles come from the components.

**Comment by thecrypticace** [comment]
Oh you mean using `globalStyles` in conjunction with `extractStyles`?

**Comment by mikerogerz** [comment]
@thecrypticace Exactly, which worked fine before (unless it was an unintentional before to be in that order):


```
extractStyles: 'static/landing/css/landing.css',
globalStyles: 'assets/sass/global.scss'
```

**Comment by thecrypticace** [comment]
Global styles should absolutely be _first_ in that. They're primarily intended for use in things like sass / less to define global variables which is why they need to be first. 🤔
